client/src/utils/api.js

```diff
@@ -174,3 +175,110 @@ export const deleteResource = (id: string): ApiResponse<null> => {
       },
     );
 };
+
+export const addCategory = (category: Category): ApiResponse<Category> =>
+  instance
+    .post('/api/admin/categories', category, {
+      headers: {
+        token: localStorage.getItem('token'),
+      },
+    })
+    .then(
+      (res) => res.data,
+      (err) => {
+        console.error(err);
+        return null;
+      },
+    );
+
+export const editCategory = (
+  id: string,
+  category: Category,
+): ApiResponse<Category> => {
+  const requestExtension = `/api/admin/categories/${id}`;
+  return instance
+    .put(requestExtension, category, {
```

Comment:
Can we change the name of this function to `renameCategory` to be more clear?

In addition, to stay consistent with the suggestions I made on this endpoint - I think we can just pass in `newName` and `currentName` as parameters to this function, and then in the `.put`, we can just do `instance.put(requestExtension, { newName, currentName }`